.github/workflows/ci.yml

```diff
@@ -5,9 +5,9 @@ jobs:
   test:
     strategy:
       matrix:
-        go: [ 1.19, 1.18, 1.17, 1.16 ]
+        go: [ "1.20", "1.19", "1.18", "1.17", "1.16" ]
         os: [ ubuntu-22.04, ubuntu-20.04 ]
-    name: Tests Go ${{ matrix.go }} # This name is used in main branch protection rules
+    name: Tests Go ${{ matrix.go }} on ${{ matrix.os }} # This name is used in main branch protection rules
```

Comment:
We have "Protect matching branches" for the main branch which is based on full name like "Tests Go 1.18".
If we will use a 5 `matrix.go` with a 2 `matrix.os` we need to have 10 protection rules.

I suggest supporting only the last 2 or 3 versions of Go.

Reply:
When I created this PR there were 4 Go versions. Now (since 1bd36e617e4313a6a231f0a34a8fc007cbb8aff2) there are only 3 Go versions. So instead of 4x2=8 there are 3x2=6 matrix combinations.

https://endoflife.date/go shows that only 1.20 and 1.19 are supported upstream (the release policy is on https://go.dev/doc/devel/release)